src/foundry/client-esm/applications/apps/compendium-art-config.d.mts

```diff
@@ -1,34 +1,37 @@
+import type { InterfaceToObject } from "../../../../types/helperTypes.d.mts";
 import type { DeepPartial } from "../../../../types/utils.d.mts";
 import type { CompendiumArtDescriptor } from "../../helpers/_types.d.mts";
-import type { ApplicationConfiguration, ApplicationRenderOptions, FormFooterButton } from "../_types.d.mts";
 import type ApplicationV2 from "../api/application.d.mts";
 import type HandlebarsApplicationMixin from "../api/handlebars-application.d.mts";
 
+type CompendiumArtConfigRenderContext = InterfaceToObject<CompendiumArtConfig.RenderContext>;
```

Comment:
So interfaces can't get treated like `Record<string, unknown>` because technically interfaces aren't ever safe to treat as plain objects because of declaration merging.

If someone declaration merges in a function, like:
```ts
namespace CompendiumArtConfig {
     interface RenderContext {
          (someSignature: number): void
     }
}
```

Then assigning `CompendiumArtConfig.RenderContext` to `Record<string, unknown>` is no longer sound. `InterfaceToObject` just strips this out if it were to occur. I might consider making it a compiler error with something like `MustNotConform` against a function. It wouldn't brick any other types since actual usage would just silently strip out the function parts but it'd explain the issue if anyone does happen to declaration merge in a function. But most people don't even know about this feature.

This is one of the reasons why I historically leaned away from interfaces but I realized it's:
1. More consistent to allow nearly everything to be extensible.
2. _Largely_ easier to explain to contributors "just use interface".
3. Can be more performant. It can cache and sort out some things `type`s can't.

And anyways I already figured out the thorny issues and it has to be supported in some of the most problematic areas like `DataModel` schemas already.